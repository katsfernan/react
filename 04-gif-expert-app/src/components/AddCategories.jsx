import { PropTypes } from 'prop-types';
import { useState } from 'react';

export const AddCategories = ({ onNewCategory }) => {
  
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim()
        if (newInputValue <= 1) return;

        // setCategories([...categories, inputValue])
        onNewCategory( newInputValue );
        setInputValue('');
    }

    return (
    <form onSubmit={ onSubmit } aria-label= "form">
        <input 
            type="text" 
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
        <button type='submit'>Agregar categoria</button>
    </form>

  )
}

AddCategories.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
