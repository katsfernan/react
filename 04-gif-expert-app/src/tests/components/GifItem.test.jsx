import { render , screen} from '@testing-library/react'

import { GifItem } from "../../components/GifItem";

describe('Prueba <GifItem/>', () => { 

    const title = 'Soy un titulo de un Gif'

    const url = 'https://giphy.com/gifs/callofduty-call-of-duty-modern-warfare-2-ii-eFxpuiAuG4nrPNCPEM'

    test('Debe mostrar el mismo snapshot', () => { 

        render
        const { container } = render(
            <GifItem
            title = { title }
            url = { url }
            />
        )
        expect( container ).toMatchSnapshot();
     })
 })