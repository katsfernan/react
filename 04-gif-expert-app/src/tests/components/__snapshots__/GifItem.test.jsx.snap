// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifItem/> Debe mostrar el mismo snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un titulo de un Gif"
      src="https://giphy.com/gifs/callofduty-call-of-duty-modern-warfare-2-ii-eFxpuiAuG4nrPNCPEM"
    />
    <p>
      Soy un titulo de un Gif
    </p>
  </div>
</div>
`;
